src/IO/Archives/createArchiveWriter.cpp

```diff
@@ -19,16 +20,24 @@ std::shared_ptr<IArchiveWriter> createArchiveWriter(const String & path_to_archi
 }
 
 
-std::shared_ptr<IArchiveWriter> createArchiveWriter(
-    const String & path_to_archive,
-    [[maybe_unused]] std::unique_ptr<WriteBuffer> archive_write_buffer)
+std::shared_ptr<IArchiveWriter>
+createArchiveWriter(const String & path_to_archive, [[maybe_unused]] std::unique_ptr<WriteBuffer> archive_write_buffer)
 {
     if (path_to_archive.ends_with(".zip") || path_to_archive.ends_with(".zipx"))
     {
 #if USE_MINIZIP
         return std::make_shared<ZipArchiveWriter>(path_to_archive, std::move(archive_write_buffer));
 #else
         throw Exception(ErrorCodes::SUPPORT_IS_DISABLED, "minizip library is disabled");
+#endif
+    }
+    //todo add support for extentsions i.e .gz
```

Comment:
how hard is it to do?
also what about 7z?

Reply:
Both amount to making the correct libarchive calls in the `LibArchiveWriter` constructor.

Extensions are is easy. But this would create a potential difference in how compression options are passed when creating backups. For zips you use the setting `SETTINGS compression_method='lzma'` where as for a tar you would want to do `BACKUP TABLE test.table TO Disk('backups', '1.tar.gz')`   

This does bring up the interesting discussion of how much performing compression on the archives actually reduces backup size. In my experience, the size difference can be negligible.   

7zip may be slightly harder (but maybe a good starter project?).

Reply:
I see, compression_method would be confusing but I'm okay initially just having it make decision based on the extension, same as for reading from archive.

> This does bring up the interesting discussion of how much performing compression on the archives actually reduces backup size.

Depends on the data probably, but I rarely see tar used without compressions so I would assume users would automically use something like `tar.gz` if we say it works with `tar`.

> 7zip may be slightly harder (but maybe a good starter project?).

But shouldn't it be same as for `tar`? I assumed `libarchive` has unfiorm API for all archive types.
Same as for supporting different compressions methods for tar.

Reply:
I will add support for .gz extensions and others.

>But shouldn't it be same as for tar? I assumed libarchive has unfiorm API for all archive types.
Same as for supporting different compressions methods for tar.

Yes. libarchive has a uniform api. I don't know a ton about 7zip, but let me see if I can just plug it in.

Reply:
it looks like 7zip is gonna take a bit of extra work to get `LibArchiveReader` working when a `ReadBuffer` is passed in (rather than assuming the archive is on disk). I am going to leave this for a future PR.